README.md

```diff
@@ -39,29 +59,106 @@ About tcsh:
 
-Dependencies:
+**Dependencies:**
   
-  1. 64 Machine with 64 bit Linux or OS X
-  2. CMake (minimum v3.15)
-  3. Python 3 and above
-  4. Python YAML library (PyYAML)
-  5. Clang v13.0 (commit: 23dd750279c9)
-  6. LLVM v13.0 (commit: 23dd750279c9), built with CMake
-    * Build llvm-13.0 **WITH CMAKE** using flag `-DLLVM_REQUIRES_RTTI=1`. [Instructions](http://llvm.org/docs/CMake.html)
-    * Remember to run `make` in the llvm build directory after running `cmake`.
-  9. GraphViz package (for visualizing error propagation)
+  1. 64 Bit Machine (preferably with GPU for faster training of ML programs) 
+  2. 64 bit Linux (Ubuntu 20.04) or OS X
+  3. CMake (minimum v3.15)
+  4. Python 3 and above
+  5. Python YAML library (PyYAML)
+  6. Ninja >= 1.10.2
+  7. libprotoc >= 3.11.0
+  8. Clang v13.0 (commit: 23dd750279c9)
+  9. LLVM v13.0 (commit: 23dd750279c9) ([Reference](http://llvm.org/docs/CMake.html)).
+		LLVM 13.0 takes a long time to completely build. Following is a shortcut to checking out the required LLVM commit, and building only the necessary LLVM targets.
+		```
+		git clone https://github.com/llvm/llvm-project.git
+		
+		# Check out a specific branch that is known to work with ONNX MLIR.
+		cd llvm-project && git checkout 23dd750279c9 && cd ..
+			
+		mkdir llvm-project/build
+		cd llvm-project/build
+		
+		cmake -G Ninja ../llvm \
+			-DLLVM_ENABLE_PROJECTS="clang;mlir;tools" \
+			-DLLVM_BUILD_TESTS=ON \
+			-DLLVM_TARGETS_TO_BUILD="host" \
+			-DLLVM_ENABLE_ASSERTIONS=ON \
+			-DLLVM_ENABLE_RTTI=ON
+
+		cmake --build . --target clang check-mlir mlir-translate opt llc lli llvm-dis llvm-link
+
+		ninja install
+		```
+  10. For executing ML programs, following additional dependencies have to be installed:
+		1. TensorFlow framework (v2.0 or greater)
+		2. numpy package (part of TensorFlow)
+		3. [tensorflow-onnx](https://github.com/onnx/tensorflow-onnx): 
+		    Installation with pip is sufficient
+		    ```
+		    pip install tf2onnx
+		    ```
+		4. libprotoc
+			```
+			curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.17.2/protobuf-all-3.17.2.zip
+			unzip protobuf-all-3.17.2.zip
+			cd protobuf-3.17.2
+			
+			./configure
+			make
+			make check
+			sudo make install
+			sudo ldconfig # refresh shared library cache.
+			```
+		5. [ONNX-MLIR](https://github.com/onnx/onnx-mlir)
+
+		    The MLIR_DIR cmake variable must be set before building onnx-mlir. It should point to the mlir cmake module inside an llvm-project build or install directory (e.g., llvm-project/build/lib/cmake/mlir).
+            ```
+		    MLIR_DIR=$(pwd)/llvm-project/build/lib/cmake/mlir
+		    ```
+
+		    Onnx-mlir commit: ``` 221b8e1d2ad ``` has to be built and installed. 
+			```
+			git clone --recursive https://github.com/onnx/onnx-mlir.git
+			cd onnx-mlir && git checkout 221b8e1d2ad && cd ..
+	
+			mkdir onnx-mlir/build && cd onnx-mlir/build
+			if [[ -z "$pythonLocation" ]]; then
+			cmake -G Ninja \
```

Comment:
Remove the if-else condition. We only need the if part. Just like we have done in the Dockerfile.